### ISSUE TITLE
can't import journal on macos

### ISSUE THREAD

**Issue by bhawkins** [pyre/pyre#82]
The following works fine:
```
$ python3 -c "import pyre; import journal"
```
But trying to import journal alone throws an `AttributeError`
```
$ python3 -c "import journal"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/bhawkins/tools/isce3/packages/journal/__init__.py", line 10, in <module>
    from . import exceptions
  File "/Users/bhawkins/tools/isce3/packages/journal/exceptions.py", line 8, in <module>
    from pyre.framework.exceptions import PyreError
  File "/Users/bhawkins/tools/isce3/packages/pyre/__init__.py", line 255, in <module>
    executive = boot()
  File "/Users/bhawkins/tools/isce3/packages/pyre/__init__.py", line 187, in boot
    return framework.executive().boot()
  File "/Users/bhawkins/tools/isce3/packages/pyre/framework/Executive.py", line 548, in boot
    self.initializeNamespaces()
  File "/Users/bhawkins/tools/isce3/packages/pyre/framework/Executive.py", line 637, in initializeNamespaces
    self.fileserver.initializeNamespace()
  File "/Users/bhawkins/tools/isce3/packages/pyre/framework/FileServer.py", line 170, in initializeNamespace
    self[self.STARTUP_DIR] = self.retrieveFilesystem(root=startup)
  File "/Users/bhawkins/tools/isce3/packages/pyre/framework/FileServer.py", line 364, in retrieveFilesystem
    folder = self.local(root=root).discover(levels=levels)
  File "/Users/bhawkins/tools/isce3/packages/pyre/filesystem/Local.py", line 240, in discover
    channel = journal.debug("pyre.filesystem.discover")
AttributeError: partially initialized module 'journal' has no attribute 'debug' (most likely due to a circular import)
```

The following provides details on system, python, and package versions:
```
$ uname -v
Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64
$ python3
Python 3.10.6 (main, Aug 30 2022, 05:12:36) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyre
>>> pyre.version()
(1, 11, 2, '132482c22')
```

**Comment by aivazis** [comment]
Does this fail regardless of what current working directory? Can you run `ls -l` when it fails?

It appears that while exploring the current working directory, the file type recognizer has bumped into a file where `stat` fails. To speed hunting the file down, you could uncomment line 234 in `packages/pyre/filesystem/Local.py` so we can see its filename.

The circular import message is a bit of a red herring. `pyre` and `journal` are carefully interwove since they depend on each other. Failures that happen early in the bootstrapping occasionally bump into this, caused mostly by surprising things in user land, but there are so few of them left that i haven't spent the time to track them all down. This failure looks like on of them.

**Comment by bhawkins** [comment]
Thanks! It doesn't fail in a new empty directory. In my home directory it doesn't like some Excel backup file:
```
$ ls -l '~$uavsar_flight_currency.xlsx'
-rw-r--r--@ 1 bhawkins  staff  165 Aug  2 12:11 ~$uavsar_flight_currency.xlsx
$ file '~$uavsar_flight_currency.xlsx'
~$uavsar_flight_currency.xlsx: data
```
It seems like any file whose name starts with `~` triggers the failure.

**Comment by aivazis** [comment]
The `@` in the file listing means that the file has extended attributes. I wonder whether that has something to do with the problem. I'll try to make some of these files and see what's causing the failure.

**Comment by aivazis** [comment]
Could you please verify that

```
import os
os.stat("~$uavsar_flight_currency.xlsx")
```

raises a `FileNotFoundError`? If this is the case, the change i just pushed will probably fix the problem

**Comment by bhawkins** [comment]
I already deleted the file, so I can't say for sure what the extended attributes were or if `os.stat` would complain. However, I was able to inspect it with `file` and open it with a text editor. Like I said, just a tilde was sufficient to trigger the problem:
```
$ echo hello > '~world'
$ ls -l '~world'
-rw-r--r--  1 bhawkins  staff  6 Sep 13 09:59 ~world
$ python3 -c "import journal" &>/dev/null || echo failed
failed
```
Maybe some code was trying to expand '~' into a home directory or something. Anyways, it appears the logic in 029af6fa is sufficiently general to solve the problem. Thanks!
